File: StormEngine/src/main/scala/com/pastorm/encounter/dice/Die.scala
package com.pastorm.encounter.dice

import java.security.SecureRandom

class Die (val sides:Int) extends Rollable[Int] {

  val randy = new SecureRandom()
  def roll:Int = randy.nextInt(sides) + 1
  override def toString:String = "d" + sides
}

File: DDModelLibrary/src/main/java/com/ddmodel/Die.java
package com.ddmodel;

import java.security.SecureRandom;

public class Die {
    private int sides;
    private final SecureRandom randy = new SecureRandom();

    public Die(int sides) {
        this.sides = sides;
    }

    public int roll() {
        return randy.nextInt(sides) + 1;
    }

    @Override
    public String toString() {
        return "d" + sides;
    }
}
